main.cpp

```diff
@@ -746,6 +857,38 @@ int main()
 {
     Example::main();
     
```

Comment:
You need to instantiate your nested classes in `main()`.
```
 3) Instantiate 1 or 2 instances of EACH of your user-defined types in the main() function.  
    You should have at least 12 different variables declared in main(), because you have written 12 different types (including the nested types)
```
- [ ] revise